src/main/java/com/sngular/kloadgen/sampler/SamplerUtil.java

```diff
@@ -370,9 +377,12 @@ public static BaseLoadGenerator configureKeyGenerator(final Properties props) {
     final BaseLoadGenerator generator;
 
     final String keyNameStrategy = jMeterVariables.get(ProducerKeysHelper.KEY_NAME_STRATEGY);
-
     if (Objects.isNull(keyNameStrategy)) {
-      props.put(ProducerKeysHelper.KEY_NAME_STRATEGY, ProducerKeysHelper.TOPIC_NAME_STRATEGY);
+      if (SchemaRegistryKeyHelper.SCHEMA_REGISTRY_APICURIO.equalsIgnoreCase(jMeterVariables.get(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME))) {
```

Comment:
Like the comment in L.129:
SCHEMA_REGISTRY_APICURIO Vs CHEMA_REGISTRY_CONFLUENT
TOPIC_NAME_STRATEGY_APICURIO Vs TOPIC_NAME_STRATEGY_CONFLUENT

Reply:
As the answer to your comment in L.129